### ISSUE TITLE
Binaries are invalid when compiled on Java 11 and run on Java 8

### ISSUE THREAD

**Issue by lhotari** [apache/bookkeeper#4202]
**BUG REPORT**

***Describe the bug***

The current Bookkeeper maven build will produce invalid binaries for Java 8 when compiled on Java 11.

Here's an example of the problem:
```
2024-02-07T15:33:44,901 - ERROR - [BookieJournal-3181:BookieCriticalThread@41] - Uncaught exception in thread BookieJournal-3181 and is exiting!
java.lang.NoSuchMethodError: java.nio.ByteBuffer.clear()Ljava/nio/ByteBuffer;
        at org.apache.bookkeeper.bookie.JournalChannel.writeHeader(JournalChannel.java:255) ~[org.apache.bookkeeper-bookkeeper-server-4.17.0-SNAPSHOT.jar:4.17.0-SNAPSHOT]
        at org.apache.bookkeeper.bookie.JournalChannel.<init>(JournalChannel.java:188) ~[org.apache.bookkeeper-bookkeeper-server-4.17.0-SNAPSHOT.jar:4.17.0-SNAPSHOT]
        at org.apache.bookkeeper.bookie.JournalChannel.<init>(JournalChannel.java:124) ~[org.apache.bookkeeper-bookkeeper-server-4.17.0-SNAPSHOT.jar:4.17.0-SNAPSHOT]
        at org.apache.bookkeeper.bookie.Journal.run(Journal.java:996) ~[org.apache.bookkeeper-bookkeeper-server-4.17.0-SNAPSHOT.jar:4.17.0-SNAPSHOT]
```

***To Reproduce***

1. Compile binaries on Java 11.
2. Run bookkeeper on Java 8. 

***Expected behavior***

The binaries should be Java 8 compatible as long as we provide the support.

***Additional context***

Similar problem was solved in Pulsar by properly using maven.compiler.release setting.

**Comment by lhotari** [comment]
I'm handling this in #4197 since this is required for fixing CI.

**Comment by lhotari** [comment]
similar as https://github.com/apache/pulsar/issues/8445

**Comment by lhotari** [comment]
This is fixed in #4197 since it was needed to get CI passing.
